### PR TITLE
gRPC: support @Blocking on class

### LINKED ISSUE #17475: Make `@Blocking` work on gRPC services
Currently adding `@Blocking` works only on methods, it should also work on whole classes

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #17475

### CODE REVIEW

**Comment by geoand** [comment]
Dump question: Is `@NonBlocking` supported? I mean can a user add `@Blocking` on a class and `@NonBlocking` on a method?

**Comment by michalszynkiewicz** [comment]
I didn't know we had `@NonBlocking`, I'll add support for it

**Comment by geoand** [comment]
Great, thanks!

**Comment by mkouba** [comment]
> I didn't know we had `@NonBlocking`, I'll add support for it

I didn't know about this annotation either ;-)

Otherwise the PR looks good.

**Comment by mkouba** [comment]
@michalszynkiewicz @gsmet I think that this one should be backported in 2.0.Final. WDYT?

**Comment by michalszynkiewicz** [comment]
updated to support `@NonBlocking`, and added a test for it.
👍🏻  on backport

File: extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java [hunk 1]
```diff
@@ -193,13 +191,30 @@ void discoverBindableServices(BuildProducer<BindableServiceBuildItem> bindables,
...
-                if (method.hasAnnotation(GrpcDotNames.BLOCKING)) {
-                    item.registerBlockingMethod(method.name());
+                if (!BLOCKING_SKIPPED_METHODS.contains(method.name())
+                        && !method.hasAnnotation(GrpcDotNames.NON_BLOCKING)) {
+                    result.add(method);
                 }
             }
-            bindables.produce(item);
         }
+        for (MethodInfo method : service.methods()) {
```
Comment (mkouba) [inline extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java:212]: Do we have to iterate over `service.methods()` twice?

Comment (michalszynkiewicz) [reply extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java:212]: nope, it's either a bit more complex if or two iterations

**Comment by geoand** [comment]
It should certainly be backported to Final

**Review by mkouba — APPROVED** [review]
The PR looks good.  I've sent a PR with minor change: https://github.com/michalszynkiewicz/quarkus/pull/10 ;-)

**Pull request merged**